### PR TITLE
MOSIP-44717 - Automated Scan feature related testcases and framework optimization

### PR DESCRIPTION
**Pull request by mohanachandran-s**
MOSIP-44717 - Automated Scan feature related testcases and framework optimization

## Summary by CodeRabbit

* **New Features**
  * Enhanced scan/upload testing: camera-permission flows, multiple camera/profile variants, runtime test‑media generation for scan/upload, wrong‑URL checks, and upload size‑boundary cases (10KB / 5MB).

* **Bug Fixes**
  * Improved reliability with explicit waits, retries, concurrent-safe test state, and clearer success/error reporting (local screenshots vs remote session/video links).

* **Documentation**
  * Expanded UI test guide with local/BrowserStack/IDE/container run instructions, config options, and troubleshooting.

* **Tests**
  * New/updated scenarios, smoke/negative cases, and parallel test execution enabled.

* **Chores**
  * Added BrowserStack/config options and test infra tweaks; ignore rule for “.err” files.

### CODE REVIEW

File: ui-test/src/main/java/constants/UiConstants.java [hunk 1]
```diff
@@ -50,5 +50,15 @@ public class UiConstants {
...
 	public static final String ERROR_MULTI_FORMAT = "No QRCode found";
-	
-}
\ No newline at end of file
+	public static final String HALF_QRCODE_FAILURE_MESSAGE = "Session expired. Please Scan again.";
+
+	public static final String SCAN_QR_CODE_LABEL_FONT_AFTER_STEP = "ml-[10px] text-lgNormalTextSize font-bold text-stepperLabel";
+	public static final String CLASS = "class";
+	public static final String NO_INTERNET_TITLE = "No Internet Connection!";
+	public static final String NO_INTERNET_DESCRIPTION = "Oops! We cant seem to connect. Check your internet connection and try again.";
```
[the remark this answers is not in the document]
Comment (mohanachandran-s) [reply ui-test/src/main/java/constants/UiConstants.java:58]: Ignore

File: ui-test/src/main/java/constants/UiConstants.java [hunk 2]
```diff
@@ -40,7 +40,7 @@ public class UiConstants {
 	public static final String VERIFICATION_PROG = "Verification in Progress";
 	public static final String VERIFICATION_PROG_DETAIL = "This verification will take sometime, please don’t close the browser.";
 	public static final String RESULT_STRING = "Results";
-	public static final String CONGRATULATIONS_MESSAGE = "Congratulations, the given credential is valid!";
+	public static final String VERIFICATION_SUCCESS_MESSAGE = "Congratulations, the given credential is valid!";
```
[the remark this answers is not in the document]
Comment (mohanachandran-s) [reply ui-test/src/main/java/constants/UiConstants.java:43]: CONGRATULATIONS_MESSAGE no longer exists in UiConstants.java and there are zero references to it anywhere in the codebase. The constant was already cleaned
  up — only VERIFICATION_SUCCESS_MESSAGE remains at UiConstants.java:43.

File: ui-test/src/main/java/stepdefinitions/StepDef.java [hunk 3]
```diff
@@ -1824,11 +1442,11 @@ public void uploadQrCodeFileExpiredJpeg() {
...
 	}
 
-	
+
 	@When("Upload QR code file Expired pdf")
 	public void uploadQrCodeFileExpiredPdf() {
 	    try {
-	        uploadqrcode.clickOnUploadExpiredQRCodepngExpired(); 
+	        uploadqrcode.clickOnUploadExpiredQRCodepngExpired();
 	        Assert.assertTrue(true, "Expired PDF QR code uploaded successfully.");
```
[the remark this answers is not in the document]
Comment (mohanachandran-s) [reply ui-test/src/main/java/stepdefinitions/StepDef.java:1478]: already fixed

File: ui-test/src/main/java/utils/BaseTest.java [hunk 4]
```diff
@@ -207,28 +276,157 @@ public void afterScenario(Scenario scenario) {
...
 			ExtentReportManager.getTest().fail("❌ Scenario Failed: " + scenario.getName());
 		} else {
-			passedCount++;
+			passedCount.incrementAndGet();
 			ExtentReportManager.getTest().pass("✅ Scenario Passed: " + scenario.getName());
 		}
 
+		markBrowserStackSessionStatus(scenario);
+		attachBrowserStackVideoLinkIfFailed(scenario);
 		ExtentReportManager.flushReport();
```
[the remark this answers is not in the document]
Comment (mohanachandran-s) [reply ui-test/src/main/java/utils/BaseTest.java:321]: Resolved

File: ui-test/src/main/java/utils/BaseTestUtil.java [hunk 5]
```diff
@@ -1,32 +1,613 @@
...
+
+    private boolean isRunningFromIde() {
+        String classPath = System.getProperty("java.class.path", "").toLowerCase();
+        String command = System.getProperty("sun.java.command", "").toLowerCase();
+
+        return classPath.contains("idea_rt.jar")
+                || classPath.contains("eclipse")
+                || classPath.contains("junit")
+                || command.contains("com.intellij")
+                || command.contains("org.eclipse");
```
[the remark this answers is not in the document]
Comment (mohanachandran-s) [reply ui-test/src/main/java/utils/BaseTestUtil.java]: Resolved

File: ui-test/src/main/java/utils/FinalResultListener.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class FinalResultListener implements ITestListener, ISuiteListener {
+
+    private static final Logger LOGGER = Logger.getLogger(FinalResultListener.class);
+    private static final AtomicInteger passedCount = new AtomicInteger(0);
+    private static final AtomicInteger failedCount = new AtomicInteger(0);
+    private static final AtomicInteger skippedCount = new AtomicInteger(0);
+    private static final AtomicInteger totalCount = new AtomicInteger(0);
```
[the remark this answers is not in the document]
Comment (mohanachandran-s) [reply ui-test/src/main/java/utils/FinalResultListener.java:18]: Resolved

File: ui-test/src/main/resources/config/injiVerify.properties [hunk 7]
```diff
@@ -32,42 +45,42 @@ db-server = released.mosip.net
...
+mosip_resident_client_secret = MiTOjWLVoFZw3CHZ
+mosip_idrepo_client_secret = qKNTW5CTxKsgVGjw
+mosip_reg_client_secret = Msc7sItq6Rg8vcEC
+mosip_admin_client_secret = auWjWiniqX0YX8KF
+mosip_hotlist_client_secret = lA7pCSLknLvT9Owj
+mosip_regproc_client_secret = rR5eKHfG4i3BJf8S
+mpartner_default_mobile_secret = duo1HolnpqNc6aMA
+mosip_testrig_client_secret = G9qAJ3SrOm575GqL
+AuthClientSecret = MiTOjWLVoFZw3CHZ
+mosip_crvs1_client_secret =
```
[the remark this answers is not in the document]
Comment (mohanachandran-s) [reply ui-test/src/main/resources/config/injiVerify.properties]: Resolved

**Review by santhoshsunder — APPROVED** [review]

**Review by mayuradesh — APPROVED** [review]
